src/find-options/FindOptionsUtils.ts

```diff
@@ -155,7 +155,14 @@ export class FindOptionsUtils {
         if (options.lock) {
             if (options.lock.mode === "optimistic") {
                 qb.setLock(options.lock.mode, options.lock.version);
-            } else if (options.lock.mode === "pessimistic_read" || options.lock.mode === "pessimistic_write" || options.lock.mode === "dirty_read" || options.lock.mode === "pessimistic_partial_write" || options.lock.mode === "pessimistic_write_or_fail") {
+            } else if (
+                options.lock.mode === "pessimistic_read" ||
```

Comment:
Is it better to store all possible locks in const array and call `array.includes()` instead of chaining equality checks.

Reply:
Absolutely. The locking code base seems to have grown a lot in the past and needs some refactoring. I would like to create a separate PR for cleaning up. This is more like a "hot fix"